### PR TITLE
Changes default Petstore version to v3 (OAS 3.0)

### LINKED ISSUE #3208: Default Petstore specification version loaded is still Swagger 2.0 (OAS 2.0). Can we promote latest support spec version (OAS 3.0)
When loading `swagger-editor` in a fresh browser (e.g. no-cached specification document), the default specification renders is a Swagger 2.0 (OAS 2.0) version of the Petstore API. 

image

We should be promoting the latest version of the OpenAPI specification supported by the tooling chain.

### PR DESCRIPTION
**Pull request by frankkilcommins**
Changes the default spec version loaded when no-cache available to be and OAS 3.0.3 version of the Petstore API. 
Added two new menu options to allow loading of:
- Petstore OAS 3.0
- Petstore OAS 2.0
Modified how the local-storage plugin loads the default spec, to reuse same mechanism as new menu options.
Modified package.json to modify scripts to overcome flaky cypress tests that was expecting spec loading via URL but it was falling back to local-storage (please check this part as `cy:ci` may in what ever jobs its being used)

### Description
Changes the Petstore.js spec version loading in the local-storage plugin and add new menu options to allow default OAS 3.0 and OAS 2.0 Petstore definitions to be loaded into editor.

### Motivation and Context
We should be progressively enhancing the experience of the editor to support the latest compatible version of OAS

Fixes #3208

### How Has This Been Tested?
Added cypress test to validate the new menu option spec loading.
Manually via Chrome/Edge/Firefox. 
Validate the try it behaviour (note I did spot a issue with some of the Petstore v3 endpoints for error scenarios which i'll fix over there separately)
Tested code-gen - client + server for c#

### My PR contains... 

- [ ] No code changes (`src/` is unmodified: changes to documentation, CI, metadata, etc.)
- [ ] Dependency changes (any modification to dependencies in `package.json`)
- [x] Bug fixes (non-breaking change which fixes an issue)
- [ ] Improvements (misc. changes to existing features)
- [ ] Features (non-breaking change which adds functionality)

### My changes...
- [ ] are breaking changes to a public API (config options, System API, major UI change, etc).
- [ ] are breaking changes to a private API (Redux, component props, utility functions, etc.).
- [ ] are breaking changes to a developer API (npm script behavior changes, new dev system dependencies, etc).
- [x] are not breaking changes.

### Documentation
- [x] My changes do not require a change to the project documentation.
- [ ] My changes require a change to the project documentation.
- [ ] If yes to above: I have updated the documentation accordingly.

PR-OAS3-Inform how to load OAS2

PR-New Menu Items

### CODE REVIEW

**Review by frankkilcommins — COMMENTED** [review]
Updating default specification defintion loading to be an OAS3.0.3 version of the Petstore v3 API

**Review by char0n — CHANGES_REQUESTED** [review]
I've generated a following git patch to make the e2e npm scripts right. Can you please incorporate it into the PR?

```
From 404e1b056a0f6fa91e6471741777d239d26ecaf1 Mon Sep 17 00:00:00 2001
From: Vladimir Gorej <vladimir.gorej@gmail.com>
Date: Mon, 18 Jul 2022 11:36:15 +0200
Subject: [PATCH] chore: consolidate e2e npm scripts

---
 dev-helpers/index.html |   2 +-
 package.json           |   6 +-
 webpack/e2e.babel.js   | 133 +++++++++++++++++++++++++++++++++++++++++
 3 files changed, 137 insertions(+), 4 deletions(-)
 create mode 100644 webpack/e2e.babel.js

diff --git a/dev-helpers/index.html b/dev-helpers/index.html
index e3354567..684f9204 100644
--- a/dev-helpers/index.html
+++ b/dev-helpers/index.html
@@ -54,7 +54,7 @@
       presets: [
         SwaggerEditorStandalonePreset
       ],
-        queryConfigEnabled: true,
+      queryConfigEnabled: true,
     })
 
     window.editor = editor
diff --git a/package.json b/package.json
index 780db634..a8d051f6 100644
--- a/package.json
+++ b/package.json
@@ -37,14 +37,14 @@
     "lint-errors": "eslint --cache --no-error-on-unmatched-pattern --quiet --ext '.js,.jsx' src test",
     "lint-fix": "eslint --cache --ext '.js,.jsx' src test --fix",
     "cy:start": "webpack serve --port 3260 --host 0.0.0.0 --config webpack/dev.babel.js --hot --progress --static test/e2e/static",
+    "cy:start:ci": "webpack serve --port 3261 --host 0.0.0.0 --config webpack/e2e.babel.js --progress --static test/e2e/static",
     "cy:open": "cypress open",
     "cy:run:chrome": "cypress run --browser chrome",
     "cy:run:firefox": "cypress run --browser firefox",
     "cy:run:electron": "cypress run --browser electron",
     "cy:dev": "cross-env CYPRESS_BASE_URL=http://localhost:3260/ start-server-and-test cy:start http://localhost:3260 cy:open",
-    "cy:dev-e2e": "cross-env CYPRESS_BASE_URL=http://localhost:3260/ start-server-and-test cy:start http://localhost:3260 cy:run:electron",
-    "cy:ci": "cross-env CYPRESS_BASE_URL=http://localhost:3001/ start-server-and-test build:serve http://localhost:3001 cy:run:electron",
-    "test": "run-s test:unit-mocha test:unit-jest cy:dev-e2e lint-errors",
+    "cy:ci": "cross-env CYPRESS_BASE_URL=http://localhost:3261/ start-server-and-test cy:start:ci http://localhost:3261 cy:run:electron",
+    "test": "run-s test:unit-mocha test:unit-jest cy:ci lint-errors",
     "test-in-node": "run-s lint-errors test:unit-mocha test:unit-jest",
     "test:unit-jest": "cross-env BABEL_ENV=test jest --config ./config/jest/jest.unit.config.js",
     "test:unit-mocha": "cross-env BABEL_ENV=test mocha --require test/mocha/setup.js --recursive --require @babel/register test/mocha",
diff --git a/webpack/e2e.babel.js b/webpack/e2e.babel.js
new file mode 100644
index 00000000..132b5b17
--- /dev/null
+++ b/webpack/e2e.babel.js
@@ -0,0 +1,133 @@
+/**
+ * @prettier
+ */
+
+/**
+ * The goal of this config is to mimic the production build setup as close as possible for e2e testing.
+ */
+
+import path from "path"
+import HtmlWebpackPlugin from "html-webpack-plugin"
+import { HtmlWebpackSkipAssetsPlugin } from "html-webpack-skip-assets-plugin"
+
+import configBuilder from "./_config-builder"
+import styleConfig from "./stylesheets.babel"
+
+const projectBasePath = path.join(__dirname, "../")
+
+const emitWorkerAssets = false
+
+const e2eConfig = configBuilder(
+  {
+    minimize: true,
+    mangle: true,
+    sourcemaps: true,
+    includeDependencies: true,
+    emitWorkerAssets,
+  },
+  {
+    mode: "production",
+    entry: {
+      "swagger-editor-bundle": [
+        "./src/index.js",
+      ],
+      "swagger-editor-standalone-preset": [
+        "./src/standalone/index.js",
+      ],
+      "swagger-editor": "./src/styles/main.less",
+    },
+
+    performance: {
+      hints: false,
+    },
+
+    output: {
+      filename: "[name].js",
+      chunkFilename: "[id].js",
+      globalObject: "this",
+      library: {
+        name: "[name]",
+        export: "default",
+      },
+      publicPath: "/",
+    },
+
+    devServer: {
+      allowedHosts: "all", // for development within VMs
+      headers: {
+        "Access-Control-Allow-Origin": "*",
+        "Access-Control-Allow-Methods": "*",
+        "Access-Control-Allow-Headers": "*",
+      },
+      port: 3261,
+      host: "0.0.0.0",
+      static: {
+        directory: path.resolve(projectBasePath, "dev-helpers"),
+        publicPath: "/",
+      },
+      client: {
+        logging: "info",
+        progress: true,
+      },
+    },
+
+    module: {
+      rules: [
+        {
+          test: /\.worker\.js$/,
+          use: [
+            {
+              loader: "worker-loader",
+              options: {
+                inline: emitWorkerAssets ? "fallback" : "no-fallback",
+                filename: "[name].js",
+              },
+            },
+            "babel-loader",
+          ],
+        },
+        {
+          test: /\.jsx?$/,
+          include: [
+            path.join(projectBasePath, "src"),
+            path.join(projectBasePath, "node_modules", "object-assign-deep"),
+          ],
+          loader: "babel-loader",
+          options: {
+            retainLines: true,
+            cacheDirectory: true,
+          },
+        },
+        {
+          test: /\.(txt|yaml)$/,
+          type: "asset/source",
+        },
+        {
+          test: /\.(png|jpg|jpeg|gif|svg)$/,
+          type: "asset/inline",
+        },
+      ],
+    },
+
+    plugins: [
+      new HtmlWebpackPlugin({
+        template: path.join(projectBasePath, "dev-helpers", "index.html"),
+        // excludeChunks: ["validator.worker"],
+      }),
+      new HtmlWebpackSkipAssetsPlugin({
+        skipAssets: [/swagger-editor\.js/],
+      }),
+    ].filter(Boolean),
+  }
+)
+
+// mix in the style config's plugins and loader rules
+
+e2eConfig.plugins = [...e2eConfig.plugins, ...styleConfig.plugins]
+
+e2eConfig.module.rules = [
+  ...e2eConfig.module.rules,
+  ...styleConfig.module.rules,
+]
+
+export default e2eConfig
-- 
2.25.1
```

File: package.json [hunk 1]
```diff
@@ -34,16 +34,17 @@
...
-    "lint-errors": "eslint --cache --quiet --ext '.js,.jsx' src test",
+    "lint-errors": "eslint --cache --no-error-on-unmatched-pattern --quiet --ext '.js,.jsx' src test",
     "lint-fix": "eslint --cache --ext '.js,.jsx' src test --fix",
     "cy:start": "webpack serve --port 3260 --host 0.0.0.0 --config webpack/dev.babel.js --hot --progress --static test/e2e/static",
     "cy:open": "cypress open",
     "cy:run:chrome": "cypress run --browser chrome",
     "cy:run:firefox": "cypress run --browser firefox",
     "cy:run:electron": "cypress run --browser electron",
     "cy:dev": "cross-env CYPRESS_BASE_URL=http://localhost:3260/ start-server-and-test cy:start http://localhost:3260 cy:open",
+    "cy:dev-e2e": "cross-env CYPRESS_BASE_URL=http://localhost:3260/ start-server-and-test cy:start http://localhost:3260 cy:run:electron",
```
Comment (char0n) [inline package.json:45]: Generated a git patch for making this right.

Using `build:serve` limits us in what we can do with the fragments and how we can configure to run them in e2e tests. Instead I've configured webpack and babel to mimic building of production build fragments during the e2e testing in CI.

File: package.json [hunk 2]
```diff
@@ -34,16 +34,17 @@
     "hot-server": "webpack serve --host 0.0.0.0 --config webpack/dev.babel.js",
     "open-static": "node -e 'require(\"open\")(\"http://localhost:3001\")'",
     "lint": "eslint --cache --ext '.js,.jsx' src test",
-    "lint-errors": "eslint --cache --quiet --ext '.js,.jsx' src test",
+    "lint-errors": "eslint --cache --no-error-on-unmatched-pattern --quiet --ext '.js,.jsx' src test",
```
[the remark this answers is not in the document]
Comment (frankkilcommins) [reply package.json]: done

File: dev-helpers/index.html [hunk 3]
```diff
@@ -53,7 +53,8 @@
       layout: 'StandaloneLayout',
       presets: [
         SwaggerEditorStandalonePreset
-      ]
+      ],
+        queryConfigEnabled: true,
```
[the remark this answers is not in the document]
Comment (frankkilcommins) [reply dev-helpers/index.html]: done

File: test/e2e/tests/definitionLoading.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+describe("Menu Definition Loading", () => {
```
[the remark this answers is not in the document]
Comment (frankkilcommins) [reply test/e2e/tests/definitionLoading.js]: done

**Review by char0n — APPROVED** [review]
Looks good!

**Comment by char0n** [comment]
Moved to backlog pipeline. This gets shipped during tommorow.

**Pull request merged**